### ISSUE TITLE
Error when "restore database from URL" encountered in sql files

### ISSUE THREAD

**Issue by timabell** [jmezach/MSBuild.Sdk.SqlProj#25]
I followed along with the instructions at https://erikej.github.io/efcore/2020/05/11/ssdt-dacpac-netcore.html and it was going okay until the build encountered an azure `restore database xxx from url = 'xxx'` statement, at which point it errored:

```sql
Adding /home/tim/repo/.../Restore_v1.0.sql to the model
Unhandled exception: System.Reflection.TargetInvocationException:
Exception has been thrown by the target of an invocation.
 ---> Microsoft.SqlServer.Dac.Model.DacModelException: Add or update objects failed due to the following errors:
EXEC : error SQL46010: Incorrect syntax near URL.
[/home/tim/repo/.../Database.Build.csproj]
```

The `restore database from url` option is documented here: https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/sql-server-backup-to-url?view=sql-server-ver15#support-for-restore-arguments and I see there is a matching option on `backup database to`

Thanks for the project!

**Comment by ErikEJ** [comment]
Please share your project (at least options)

**Comment by timabell** [comment]
The work in progress is here: https://github.com/SkillsFundingAgency/dfc-coursedirectory/blob/linux-builds/src/Dfc.CourseDirectory.Database.Build/Database.Build.csproj (work in progress branch, may disappear later)

It pulls in https://github.com/SkillsFundingAgency/dfc-coursedirectory/tree/cd887928fb48e08cafcd8c34eac8d207ebe283f0/src/Dfc.CourseDirectory.Database

**Comment by jmezach** [comment]
I had a look around and according to [this post](https://social.msdn.microsoft.com/Forums/en-US/aedad849-46c2-412c-8c35-b48d8fe8e4b4/sql46010-backup-to-url?forum=ssdt) it seems that it is also an issue with the existing `.sqlproj` format. @timabell Does the .sqlproj compile as it is, or does it give you the same error?

The post linked above suggests to use dynamic SQL to resolve it, although that is of course less ideal.

**Comment by ErikEJ** [comment]
Or maybe move the offending script to PostDeployment with "CREATE OR UPDATE" ?

**Comment by timabell** [comment]
@jmezach The visual studio sqlproj does compile and create a dacpac, yes.

I tried the dynamic sql option and that indeed get past the error though at a cost of the ability of tooling to analyse the file.

I've realised I probably don't actually need that script in the linux build of the dacpac for local development at the moment so I've altered my linux build csproj to not include the problematic folder and now have a successful build. Thanks for your help on this.

@ErikEJ thanks for the suggestion, I didn't get as far as trying it.

I don't know if it makes sense to keep this issue open as I'm not that familiar with dacpacs and don't know if a database restore would be a normal thing to keep in them. I may return here if I end up needing it as part of the build setup later.

Thanks again, very grateful for the quick and helpful responses.

**Comment by jmezach** [comment]
I would expect the original .sqlproj to fail with the same error actually, since both project file formats should be using the same SQL parser. Unless the offending file is not part of the build in the original .sqlproj. What is the build action for that file?

Either way I don't think there's much we can do here if the parser doesn't support it. I checked if there's an update to the NuGet package we're using, but alas there's no newer stable version.

@ErikEJ I've added the suggested workarounds to the [README](https://github.com/jmezach/MSBuild.Sdk.SqlProj#workaround-for-parser-errors-sql46010) so I think we can go ahead and close this.
